### ISSUE TITLE
Performance concern on observing all Video node fields

### ISSUE THREAD

**Issue by jeanbenitezu** [googleads/googleads-ima-roku-dai#5]
Hi Google Ads Team!

I have a concern with performance here, [this line](https://github.com/googleads/googleads-ima-roku-dai/blob/main/podserving_example/components/Sdk.xml#L42) is attaching an observer for each video node field, and they should be a lot, and even worst, some of them are content nodes, so when something on these nodes changes, it may trigger the observer. 

My question is: is it necessary in that way?

I made some small changes just to observe a limited number of fields and, it seems it works in the same way as observing all the fields

```["position", "state", "control", "timedMetaData", "timedMetaData2"]```

What do you recommend?

Thanks in advance!

**Comment by Kiro705** [comment]
Hi @jeanbenitezu ,

Sorry for the delaying in replying. Would you be able to add a code snippet demonstrating the change to the video node observer changes you are making? The link you gave pointed to an outdated Github branch.

Once I get more details I should be able to confirm if there are any expected issues with the change.

Thank you,
Jackson
IMA SDK team

**Comment by jeanbenitezu** [comment]
Actually, the change suggested was made LOL
https://github.com/googleads/googleads-ima-roku-dai/blob/main/pod_serving_example/components/Sdk.xml#L47-L50

Previous to that, it was doing a loop with all the fields of the video node.

Thanks Jackson!

**Comment by jeanbenitezu** [comment]
This is the change I made there. You can see that I kept it commented, because I was waiting for confirmation here.

```brs
' OPTION 1: Original - All video fields
fields = video.getFields()

' OPTION 2: A limited list of fields
' fields = ["position", "state", "control", "timedMetaData", "timedMetaData2"]

for each field in fields
    video.observeFieldScoped(field, m.port, ["position"])
end for
```

It is kind of similar to the new podserving example

**Comment by Kiro705** [comment]
I see, thanks for clarifying.

It should be fine to use the `OPTION 2` cases in your code snippet.

Did you have any other questions or concerns regarding the fields being observed?

**Comment by jeanbenitezu** [comment]
No, this is perfecto. Thank you, Jackson! Feel free to solve this issue.
